.circleci/config.yml

```diff
@@ -23,6 +23,30 @@ commands:
       - restore_cache:
           keys:
             - 'v2-pkg-cache-{{ checksum "go.sum" }}-{{ .Environment.GOVERSION }}'
+            - 'bin-cache-{{ .Branch }}'
+
+  setup-install-bins:
+    description: "install 3rd-party binaries for integration tests"
+    steps:
+      - setup
+      - run:
+          name: "install kubectl"
+          command: |
+            if [ ! -f ~/bin/kubectl ]; then
+            curl -L https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl -o ~/bin/kubectl
+            chmod +x ~/bin/kubectl
+            fi
+      - run:
+          name: "install kind"
+          command: |
+            if [ ! -f ~/bin/kind ]; then
+            curl -L https://github.com/kubernetes-sigs/kind/releases/download/v0.8.1/kind-linux-amd64 -o ~/bin/kind
+            chmod +x ~/bin/kind
+            fi
+      - save_cache:
+          key: 'bin-cache-{{ .Branch }}'
```

Comment:
This would get the latest kubectl, downloading it once per branch.. Maybe just get rid of the `{{ .Branch }}` and just keep everything static.

But the general idea is to add kubectl and kind binaries.